frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/update-block-pool-modal.tsx

```diff
@@ -147,8 +147,8 @@ const UpdateBlockPoolModal = withHandlePromise((props: UpdateBlockPoolModalProps
 });
 
 type UpdateBlockPoolModalProps = {
-  kind?: string;
-  blockPoolConfig: StoragePoolKind;
+  kind?: K8sKind | string;
```

Comment:
```suggestion
  kind?: K8sKind;
```
Let's only support one or the other. We can avoid a bunch of `if` statements if we choose to stick to one.

Reply:
ack